packages/zero-runtime/README.md

```diff
@@ -451,3 +503,107 @@ declare module '@mui/zero-runtime/theme' {
   }
 }
 ```
+
+## How-to guides
+
+### Coming from Emotion or styled-components
+
+Emotion and styled-components are runtime CSS-in-JS libraries. What you write in your styles is what you get in the final bundle which means that the styles can be as dynamic as you want with the trade-offs of bundle size and performance overhead.
+
+Zero-runtime, on the other hand, extracts CSS at build time and replaces the JS code with hashed classnames and some CSS variables. This means that it has to know all of the styles to be extracted ahead of time, so there are rules and limitations that you need to be aware of when using javascript callbacks or variables in zero-runtime APIs.
+
+Here are some common patterns and how to achieve them with zero-runtime:
+
+1. A fixed set of styles
+
+In Emotion or styled-components, you can use props to create styles conditionally:
+
+```js
+const Flex = styled('div')((props) => ({
+  display: 'flex',
+  ...(props.vertical // ❌ Zero Runtime will throw an error
+    ? {
+        flexDirection: 'column',
+        paddingBlock: '1rem',
+      }
+    : {
+        paddingInline: '1rem',
+      }),
+}));
+```
+
+But in zero-runtime, you need to define all of the styles ahead of time using the `variants` key:
+
+```js
+const Flex = styled('div')((props) => ({
+  display: 'flex',
+  variants: [
+    {
+      props: { vertical: true },
+      style: {
+        flexDirection: 'column',
+        paddingBlock: '1rem',
+      },
+    },
+    {
+      props: { vertical: false },
+      style: {
+        paddingInline: '1rem',
+      },
+    },
+  ],
+}));
+```
+
+> 💡 Keep in mind that `variants` is for fixed values of props, for example component's colors, sizes, states.
+
+2. Programatically generated styles
+
+For emotion and styled-components, the styles will be different on each render and instances because the styles are generated at runtime:
```

Comment:
```suggestion
For Emotion and styled-components, the styles will be different on each render and instance because the styles are generated at runtime:
```